src/main/java/no/unit/nva/doi/transformer/utils/DataciteTypesUtil.java

```diff
@@ -0,0 +1,75 @@
+package no.unit.nva.doi.transformer.utils;
+
+import no.unit.nva.doi.transformer.model.internal.external.DataciteResponse;
+import no.unit.nva.doi.transformer.model.internal.external.DataciteTypes;
+import no.unit.nva.model.PublicationType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.nonNull;
+
+public class DataciteTypesUtil {
+
+    public static final String TYPE_TEXT = "text";
+    public static final String JOURNAL = "journal";
+    public static final String ARTICLE = "article";
+    public static final int SINGLETON = 1;
+    public static final int ONLY_ELEMENT = 0;
+
+    /**
+     * Maps the potentially many content types found in a Datacite response to a PublicationType.
+     *
+     * @param dataciteResponse a DataciteResponse document.
+     * @return a PublicationType.
+     */
+    public static PublicationType mapToType(DataciteResponse dataciteResponse) {
+        DataciteTypes types = dataciteResponse.getTypes();
+        String resourceType = Optional.ofNullable(types.getResourceTypeGeneral()).orElse(null);
+        if (nonNull(resourceType) && resourceType.equalsIgnoreCase(TYPE_TEXT)) {
+            return getAnalyzedType(types);
+        }
+        return null;
+    }
+
+    private static PublicationType getAnalyzedType(DataciteTypes types) {
```

Comment:
this method needs in general to be refactored so that its logic is more apparent. Now it is easy to understand what you want to achieve